src/projects/digital/api/circuit/src/public/impl/DigitalComponentInfo.ts

```diff
@@ -1,27 +1,30 @@
+import {ErrE, Ok} from "shared/api/circuit/utils/Result";
+import {AddErrE} from "shared/api/circuit/utils/MultiError";
 import {ComponentInfoImpl} from "shared/api/circuit/public/impl/ComponentInfo";
 
-import {extend} from "shared/api/circuit/utils/Functions";
-
-import {DigitalComponentInfo as DigitalComponentInfoInternal} from "digital/api/circuit/internal/DigitalComponents";
+import {DigitalComponentConfigurationInfo} from "digital/api/circuit/internal/DigitalComponents";
 
 import {DigitalComponentInfo} from "../DigitalComponentInfo";
 
-import {DigitalCircuitState, DigitalTypes} from "./DigitalCircuitState";
-
-
-export function DigitalComponentInfoImpl(state: DigitalCircuitState, kind: string) {
-    const info = state.internal.getComponentInfo(kind).unwrap();
-    if (!(info instanceof DigitalComponentInfoInternal))
-        throw new Error(`Received non-digital component info for ${kind}!`);
-
-    const base = ComponentInfoImpl<DigitalTypes>(state, kind);
-
-    return extend(base, {
-        get inputPortGroups(): readonly string[] {
-            return info.inputPortGroups;
-        },
-        get outputPortGroups(): readonly string[] {
-            return info.outputPortGroups;
-        },
-    } as const) satisfies DigitalComponentInfo;
+import {DigitalTypes} from "./DigitalCircuitState";
+
+
+export class DigitalComponentInfoImpl extends ComponentInfoImpl<DigitalTypes> implements DigitalComponentInfo {
+    protected override getInfo() {
+        return this.state.internal.getComponentInfo(this.kind)
+            .andThen<DigitalComponentConfigurationInfo>((info) => {
+                if (!(info instanceof DigitalComponentConfigurationInfo))
+                    return ErrE("");
```

Comment:
Can you keep the error message that was thrown before?

Reply:
good catch